prime-router/src/main/kotlin/azure/ValidateFunction.kt

```diff
@@ -50,23 +51,40 @@ class ValidateFunction(
         ) request: HttpRequestMessage<String?>
     ): HttpResponseMessage {
         return try {
-            val sender: Sender?
+            val schema = request.queryParameters.getOrDefault(SCHEMA_PARAMETER, null)
+            val format = request.queryParameters.getOrDefault(FORMAT_PARAMETER, null)
             val senderName = extractClient(request)
-            sender = if (senderName.isNotBlank()) {
-                workflowEngine.settings.findSender(senderName)
+            var sender: Sender? = null
+
+            if (senderName.isNotBlank()) {
+                val clientSender = workflowEngine.settings.findSender(senderName)
                     ?: return HttpUtilities.bad(request, "'$CLIENT_PARAMETER:$senderName': unknown sender")
-            } else {
-                try {
-                    getDummySender(
-                        request.queryParameters.getOrDefault(SCHEMA_PARAMETER, null),
-                        request.queryParameters.getOrDefault(FORMAT_PARAMETER, null)
-                    )
-                } catch (e: InvalidParameterException) {
-                    return HttpUtilities.bad(request, e.message.toString())
+
+                // only use client Sender in two cases
+                // 1) only client is sent (not schema or format) - backwards compatibility
+                // 2) client is sent and matches up with provided schema and format to use Sender settings
+                if (
+                    (schema.isNullOrEmpty() && format.isNullOrEmpty()) ||
+                    (
+                        (clientSender as TopicSender).schemaName == schema &&
+                            Sender.Format.valueOf(format) == clientSender.format
+                        )
+                ) {
+                    sender = clientSender
                 }
             }
+
+            if (sender == null) {
```

Comment:
One thing I thought of here is maybe it's more idiomatic/cleaner if we replace this whole block with just `sender = sender ?: getDummySender(schema, format)` and then in the general catch at the bottom you can do a `WHEN` expression on the error type?

```
 catch (ex: Exception) {
            when (ex) {
                is InvalidParameterException -> HttpUtilities.bad(request, ex.message.toString())
                else -> {
                    if (ex.message != null) {
                        logger.error(ex.message!!, ex)
                    } else {
                        logger.error(ex)
                    }
                    HttpUtilities.internalErrorResponse(request)
                }
            }
        }
```
Only update if you think its worthwhile :)

Reply:
Oooh fantastic!  Will definitely update --thank you!